scio-tensorflow/src/main/scala/com/spotify/scio/tensorflow/TensorFlowFunctions.scala

```diff
@@ -306,77 +306,84 @@ class TFExampleSCollectionFunctions[T <: Example](val self: SCollection[T]) {
                              compression = compression)
 
   /**
-   * Save this SCollection of `org.tensorflow.example.Example` as a TensorFlow TFRecord file,
-   * along with a protobuf file representing the inferred `org.tensorflow.metadata.v0.Schema`.
-   * If you need to alter the schema, use [[inferExampleMetadata]] and pass altered schema to
-   * [[saveAsTfExampleFile()]].
+   * Saves this SCollection of `org.tensorflow.example.Example` as a TensorFlow TFRecord file.
    * @group output
    */
-  def saveAsTfExampleFile(path: String)(implicit ev: T <:< Example)
-  : (Future[Tap[Example]], Future[Tap[Schema]]) = {
+  def saveAsTfExampleFile(path: String)(implicit ev: T <:< Example): Future[Tap[Example]] = {
     this.saveAsTfExampleFile(
       path,
       schema = null,
+      schemaFilename = "_inferred_schema.pb",
+      suffix = ".tfrecords",
+      compression = Compression.UNCOMPRESSED,
+      numShards = 0
+    )(ev)
+  }
+
+  /**
+   * Saves this SCollection of `org.tensorflow.example.Example` as a TensorFlow TFRecord file,
+   * along with  `org.tensorflow.metadata.v0.Schema`.
+   * @group output
+   */
+  def saveAsTfExampleFile(path: String, schema: Schema)(implicit ev: T <:< Example)
+  : Future[Tap[Example]] = {
+    this.saveAsTfExampleFile(
+      path,
+      schema,
       schemaFilename = "_schema.pb",
       suffix = ".tfrecords",
       compression = Compression.UNCOMPRESSED,
       numShards = 0
     )(ev)
   }
 
+  /**
+   * Saves this SCollection of `org.tensorflow.example.Example` as a TensorFlow TFRecord file,
+   * along with  `org.tensorflow.metadata.v0.Schema`.
+   * @return
+   */
   def saveAsTfExampleFile(path: String,
-                          schema: SCollection[Schema],
+                          schema: Schema,
```

Comment:
This limits the use case of having Scio infer a Schema and then being able to modify it within the job (since this type change means it requires being materialized now) - is that use case important?

Reply:
This approach follows the TFX - basically what you and @yonromai suggested. In the future the flow would be:
 * schema is inferred once manually
 * schema is altered by user by hand (with some aid)
 * schema might be provided to enforce it - but not change at this point

Reply:
Got it 👍 and just to be clear when you say "manual" you just mean calling this without passing in a Schema (so it stays null) to have Scio infer it right?

Reply:
@andrewsmartin i guess there would be a separate tool, maybe based on this code maybe not, maybe it would be provided by Google and ppl would just pass the result here - we will see :)